kaitai/python/tests/utils_sbp2json.py

```diff
@@ -0,0 +1,132 @@
+# Utilities for tests cases for the python version of the Kaitai Struct SBP
+# parser only with focus on producing sbp2json-style output from Kaitai Struct
+# parser
+
+from kaitaistruct import KaitaiStream, KaitaiStruct, KaitaiStructError
+import kaitai.python.sbptable as kaitai_sbptable
+import io
+import rapidjson
+import binascii
+import sys
+import base64
+
+
+SBP_HEADER_LEN = 6
+
+
+# wrapper object which allows KaitaiStream to be used with a simple byte array
+class BufferKaitaiStream(KaitaiStream):
+    class IOBytes:
+        def __init__(self, buf):
+            self.buf = buf
+            self.pos = 0
+
+        def read(self, num):
+            if self.pos + num > len(self.buf):
+                raise EOFError
+            buf = self.buf[self.pos:self.pos + num]
+            self.pos += num
+            return buf
+
+        def seek(self, pos):
+            self.pos = pos
+
+        def tell(self):
+            return self.pos
+
+
+    def __init__(self, buf=b''):
+        super().__init__(BufferKaitaiStream.IOBytes(buf))
+
+
+    def set_buffer(self, buf):
+        self._io.buf = buf
+        self._io.pos = 0
+
+
+    def is_eof(self):
+        return True if self._io.pos == len(self._io.buf) else False
+
+
+# wrapper object which allows KaitaiStream to be used with inputs which do
+# not support seeking
+class BufferedFileKaitaiStream(BufferKaitaiStream):
+    def __init__(self, fp, max_buf_size=io.DEFAULT_BUFFER_SIZE):
+        super().__init__(fp.read(max_buf_size))
+
+        self.fp = fp
+        self.max_buf_size = max_buf_size
+
+
+    # try to read more data so that buffer size is 'max_buf_size'
+    def fill_buffer(self):
+        pos = self._io.pos
+        space = self.max_buf_size - len(self._io.buf) + pos
+        buf = self._io.buf[pos:]
+        if space > 0:
+            buf += self.fp.read(space)
+
+        self.set_buffer(buf)
+
+
+    def get_crc_bytes(self, payload_len):
+        return self._io.buf[1:SBP_HEADER_LEN + payload_len]
+
+
+# for JSON serialisation
+def serialise(obj):
+    return {k: v for k, v in obj.__dict__.items() if not k.startswith("_")}
+
+
+# convert KaitaiStruct object to have similar structure to sbp2json output
+def get_flattened_msg(obj):
+    obj.payload.preamble = ord(obj.preamble)
+    obj.payload.msg_type = obj.msg_type
+    obj.payload.sender = obj.sender
+    obj.payload.length = obj.length
+    obj.payload._io.seek(0)
+    obj.payload.payload = base64.standard_b64encode(obj.payload._io.read_bytes_full()).decode('ascii')
+    obj.payload.crc = obj.crc
+
+    return obj.payload
+
+
+# work-alike of get_next_msg_construct() based upon Kaitai Struct
+def get_next_msg_kaitai(fp):
```

Comment:
I would think we could generalize and expose it as an API for the kaitai based lib, e.g.

```
from kaitai.python import iter_messages

for msg in iter_messages(fp):
   # Do stuff with msg
```

Here we'd simply import `iter_messages` and compose it with the `get_flattened_msg` function.

Reply:
Yeah - this file is basically intended as a complete working example for anybody who wants to work with the kaitai python bindings.  What about if I move it from `kaitai/python/tests/utils_sbp2json` to `kaitai/python/tests/parser.py`?

Reply:
Sounds good